packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js

```diff
@@ -120,8 +156,13 @@ export default createConnector({
   },
 
   refine(props, searchState, event, index) {
-    if (index === undefined && this._lastReceivedPage !== undefined) {
-      index = this._lastReceivedPage + 1;
+    const pages = Object.keys(this._cachedHits || {}).map(Number);
```

Comment:
If we are storing elements with a number index in an object, what about using an array to store that instead? I guess it's no big difference either way

Reply:
I initially started with an object because user could've reached the page with params like `page=3`. Still in that case, an array can be used, though.